packages/blade/src/components/ActionList/styles/StyledListBoxWrapper.native.tsx

```diff
@@ -1,9 +1,12 @@
-import { ScrollView } from 'react-native';
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable react/display-name */
+import { SectionList } from 'react-native';
 import styled from 'styled-components/native';
 import { getBaseListBoxWrapperStyles } from './getBaseListBoxWrapperStyles';
 
 // It's not really styled here but have to keep name same since we have to style this with overflowY and maxWidth in web
-const StyledListBoxWrapper = styled(ScrollView)((props) => ({
+const StyledListBoxWrapper = styled(SectionList)((props) => ({
```

Comment:
Like what if we render `SectionList` here instead of BaseBox? https://github.com/razorpay/blade/blob/b79b2ac3f04c4f6d1ff5dd5503f6114d4c40c0ab/packages/blade/src/components/ActionList/ActionListItem.tsx#L109

Reply:
Yeah, it is styled right?

Reply:
> Like what if we render SectionList here instead of BaseBox?

We can't do that, SectionList needs the structured data of all the items to be rendered and then it does the virtualization on that data and then renders only the component which are in view. 

It won't be possible to use ActionListSection and somehow patch it to use SectionList because our ActionListSection only groups the data in a visual hierarchy, but in react-native side we only need 1 wrapper which we push all the data into. 

Right now what we do is we take the JSX and convert it to the data structure which feeds into the SectionList. 

![Frame 1483](https://user-images.githubusercontent.com/35374649/232041159-60778aaa-3b5a-470f-9b50-312157288f4a.png)

> And what is the advantage of using SectionList if we are wrapping all items in it?

So we are not wrapping all the things in SectionList, sectionList doesn't even accept any children prop, we just feed the data, and provide functions like `renderItem`, `renderSectionFooter`, `renderSectionHeader` which does the rendering.